pennylane/ops/op_math/control.py

```diff
@@ -83,6 +83,13 @@ class ControlledOperation(Operation):
         control_wires: A wire or set of wires.
         control_values: An int or list of ints indicating the values each control wire should
             take.
+
+    .. note::
+        Currently, the :func:`~.ctrl` tranform uses this class ``ControlledOperation``.  This class
+        wraps an entire :class:`pennylane.tape.QuantumTape`, and it is rarely supported for native device
+        execution.  See :class:`pennylane.ops.op_math.Controlled` for a more versatile controlled operation
+        that wraps a single target ``Operator``.

```

Comment:
What's the plan for the evolution of `ControlledOperation` and `Controlled`? It seems confusing to have both of these two classes in the long run.

Reply:
Once all of the pl controlled ops (CNOT, CZ, etc.) inherit from and are integrated with Controlled, we can then start to deprecate it !

Reply:
So `ControlledOperation` uses the `_controlled` property to decompose itself.  `Controlled` currently does not have any decompositions.  Once we add some basic `decomposition` logic to `Controlled`, I think we should prefer using it.  

Right now, `Controlled` is hidden away, but it is much more versatile, as it defines matrices, eigenvalues, diagonalizing gates, parameter shift rules, etc.